[BUGFIX] Disable init analytics as it was not really getting entirely disabled when specified on data context config

jmcorreia opened the pull request:
It seems init_analytics is being executed even when people is disabling analytics. 
Inside this execution, there seems to be a method self._get_oss_id()  searching for a root location `cls._ROOT_CONF_FILE.exists()` which is producing a permission denied error when used inside forEachBatch streaming on Databricks DBR15.4.

This PR proposes a fix to avoid invoking the init_analytics method if people disable it via config. 

If there is other suggestion for a workaround, while this PR is not release, please let me know as this is a bit critical for us.

You can find more context below:

- Closes #11275 GitHub Issue: https://github.com/great-expectations/great_expectations/issues/11275
- Discourse topic: https://discourse.greatexpectations.io/t/ephemeral-data-context-failing-with-permission-denied-on-file/2235

billdirks committed:
[BUGFIX] Make ExpectTableColumnsToMatchSet case insensitive (#11266)

wookasz committed:
[RELEASE] 1.5.3 (#11273)

jmcorreia committed:
Merge branch 'develop' into bugfix/disable-init-analytics

jmcorreia committed:
disable init analytics as it was not really getting entirely disabled

jmcorreia committed:
fix test_analytics test

wookasz reviewed (changes requested):
Thanks for the contribution. This looks reasonable. However, there are a few places in the code where `init_analytics` is invoked. Would you be willing to wrap these other calls in a similar check? If not we can follow up with that later.

Secondly, and this is where I'm requesting changes, the tests do need a little refactoring. `test_analytics_enabled_on_load` is no longer an appropriate name for the test method (it probably wasn't in the first place). Since we need to explicitly ensure that `init_analytics` is not called, there should be tests to assert that it is not called when the `expected_value` is False.

Let me know if you have any questions

jmcorreia committed:
enhance test_analytics tests and rename to capture when init_analytics is not involved.
Move try except block to cover permission denied or other errors while checking file existence

jmcorreia commented:
thanks for the feedback @wookasz . I have renamed one test and also enhanced it (and 1 more) to capture when it should be enabled and invoke init_analytics and when not. 
I did not find more places where you would need to wrap init_analytics with the if statement. In any case, I have moved the if block inside the init_analytics method so you don't need to replicate the if statement in case it needs to be used in more places.
Also moved the try except block to cover possible permission denied or other errors while checking file existence when init_analytics is enabled. 
Let me know if anything else is required.

jmcorreia committed:
Merge branch 'develop' into bugfix/disable-init-analytics

wookasz commented:
@jmcorreia `_init_analytics` is used in 3 methods in the `AbstractDataSource` class:
- `__init__`
- `enable_analytics`
- `set_user_agent_string`

The downside of moving the conditional inside `_init_analytics` is that this method is overridden in `CloudDataContext` where it also calls `_get_oss_id`

jmcorreia committed:
add check for cloud_data_context, similar to abstract data context

jmcorreia committed:
Merge remote-tracking branch 'origin/bugfix/disable-init-analytics' into bugfix/disable-init-analytics

jmcorreia committed:
Merge branch 'develop' into bugfix/disable-init-analytics

jmcorreia commented:
> @jmcorreia `_init_analytics` is used in 3 methods in the `AbstractDataSource` class:
> 
> * `__init__`
> * `enable_analytics`
> * `set_user_agent_string`
> 
> The downside of moving the conditional inside `_init_analytics` is that this method is overridden in `CloudDataContext` where it also calls `_get_oss_id`

Hi @wookasz ,
I have also changed ClodDataContext with a similar change. By changing this directly on the _init_analytics both on Abstract and Cloud, all methods that invoke it are automatically covered (init, enable_analytics and set_user_agent_string). I also feel it makes more sense for the check to be there, otherwise you would be replicating the if block in multiple places.
In any case, just let me know if you guys would prefer that approach, it just might require some other change in the tests. If you would prefer it, feel free to change, or just tell me your preference. 
We have a bit of an urgency here, so I would appreciate if this defect could have some prio.

Thanks a lot.

wookasz committed:
Update cloud_data_context.py

Skip init analytics in cloud data context
